packages/foundry/contracts/witch/WitchV2.sol

```diff
@@ -110,17 +121,44 @@ contract WitchV2 is AccessControl {
         emit LimitSet(ilkId, baseId, max);
     }
 
+    /// @dev Governance function to set other liquidation contracts that may have taken vaults already.
+    /// @param a The address that may be set/unset as another witch
+    /// @param isWitch Is this address a witch or not
+    function setAnotherWitch(address a, bool isWitch) external auth {
+        otherWitches[a] = isWitch;
+        emit AnotherWitchSet(a, isWitch);
+    }
+
+    /// @dev Governance function to to ignore pairs that can't be liquidated
+    /// @param ilkId Id of asset used for collateral
+    /// @param baseId Id of asset used for underlying
+    /// @param ignore Should this pair be ignored for liquidation
+    function setIgnoredPair(
+        bytes6 ilkId,
+        bytes6 baseId,
+        bool ignore
+    ) external auth {
+        ignoredPairs[ilkId][baseId] = ignore;
+        emit IgnoredPairSet(ilkId, baseId, ignore);
+    }
+
     /// @dev Put an undercollateralized vault up for liquidation
     /// @param vaultId Id of vault to liquidate
     function auction(bytes12 vaultId)
         external
         returns (WitchDataTypes.Auction memory auction_)
     {
-        require(auctions[vaultId].start == 0, "Vault already under auction");
-        require(cauldron.level(vaultId) < 0, "Not undercollateralized");
-
         DataTypes.Vault memory vault = cauldron.vaults(vaultId);
+        if (vault.owner == address(this) || otherWitches[vault.owner]) {
+            revert VaultAlreadyUnderAuction(vaultId, vault.owner);
+        }
         DataTypes.Series memory series = cauldron.series(vault.seriesId);
+        if (ignoredPairs[vault.ilkId][series.baseId]) {
+            revert VaultNotLiquidable(vaultId, vault.ilkId, series.baseId);
+        }
+
+        require(cauldron.level(vaultId) < 0, "Not undercollateralized");
```

Comment:
Don't you want to make this an error as well?

Reply:
I'd change all checks, but I think it should be it's own PR